fix(native): add missing description for proxy option (#6053)
getsentry/sentry-docs: src/platforms/native/configuration/transports.mdx

The transport is invoked in the same thread as the call to
`sentry_capture_event`. Consider to offload network communication to a
background thread or thread pool to avoid blocking execution.

## Using an HTTP proxy

The Native SDK allows the configuration of an HTTP proxy through which requests can be  tunneled to our backend. This proxy must allow `CONNECT` requests to establish a proxy connection. It can be configured via the following option interface:

```c
sentry_options_t *options = sentry_options_new();
sentry_options_set_http_proxy(options, "http://my.proxy:10010");
sentry_init(options);

/* ... */
```

> Both transport implementations (`Curl` and `WinHTTP`) honor the configuration on all supported platforms.

Reviewer: though we don’t forward it to the crashpad, do we?

Author: Exactly and I looked into the transport implementations of crashpad, and they explicitly all do not proxy. That should probably be an alert-box in this doc. And also something we should rectify.

Author: Another thing I thought would make sense would be to honor the `http_proxy` (etc.) environment variable(s) at some point. I also saw that some other SDKs already do that when I reviewed the basic options sections.

The transport is invoked in the same thread as the call to
`sentry_capture_event`. Consider to offload network communication to a
background thread or thread pool to avoid blocking execution.

> ## Using an HTTP proxy

Reviewer 2: ```suggestion
## Using an HTTP Proxy
```

> 
> The Native SDK allows the configuration of an HTTP proxy through which requests can be  tunneled to our backend. This proxy must allow `CONNECT` requests to establish a proxy connection. It can be configured via the following option interface:
> 

Reviewer 2: ```suggestion

The Native SDK allows the configuration of an HTTP proxy through which requests can be tunneled to our backend. This proxy must allow `CONNECT` requests to establish a proxy connection. It can be configured via the following option interface:

```

```c
sentry_options_t *options = sentry_options_new();
sentry_options_set_http_proxy(options, "http://my.proxy:10010");
sentry_init(options);

/* ... */
```

> <Alert level="warning" title="Crashpad Handler Does Not Support HTTP Proxies">
> While both transport implementations (`Curl` and `WinHTTP`) honor the configuration on all supported platforms, the `crashpad-handler` does not provide a proxy configuration and will thus not be able to send crash reports from behind a proxy.
> </Alert>

Reviewer 2: ```suggestion
<Alert level="warning" title="Note">
While both transport implementations (`Curl` and `WinHTTP`) honor the configuration on all supported platforms, the `crashpad-handler` doesn't provide a proxy configuration and will be unable to send crash reports from behind a proxy.
</Alert>
```
